app/src/python/user-scripts/gamma.py

```diff
@@ -30,17 +33,107 @@
     input_dir, '{}*.dcm'.format(evaluation_tag)))[0]
 
 
+reference = pydicom.dcmread(first_reference_dicom_file, force=True)
+evaluation = pydicom.dcmread(first_evaluation_dicom_file, force=True)
+
 gamma = gamma_dicom(
-    first_reference_dicom_file, first_evaluation_dicom_file, **gamma_options)
+    reference, evaluation, **gamma_options)
 
 valid_gamma = gamma[~np.isnan(gamma)]
 
 plt.figure()
-plt.hist(valid_gamma, 30)
+num_bins = (
+    gamma_options['interp_fraction'] * gamma_options['max_gamma'])  # type: ignore
+bins = np.linspace(0, gamma_options['max_gamma'], num_bins + 1)
+
+plt.hist(valid_gamma, bins, density=True)
 plt.xlim([0, gamma_options['max_gamma']])
 
 pass_ratio = np.sum(valid_gamma <= 1) / len(valid_gamma)
 
 plt.title("Gamma Percent Pass: {0:.2f} %".format(pass_ratio*100))
 
-plt.show()
+plt.savefig(os.path.join(output_dir, "gamma_histogram.png"))
+
+
+# (z_ref, y_ref, x_ref), dose_reference = zyx_and_dose_from_dataset(reference)
+# (z_eval, y_eval, x_eval), dose_evaluation = zyx_and_dose_from_dataset(evaluation)
+
```

Comment:
I set about plotting all the slices, but without using `random_subset`, it's just way too slow in the browser.

Reply:
I see - the idea being you don't want to use `random_subset` for plotting?

Could use `gamma_filter` instead (for demo purposes at least)?

Reply:
Fair point. Could use filter, as long as I only present pass or fail pixels...

Reply:
Won't change that in time for the workshop however, I mean we could make a second script there that uses filter instead of shell...

Reply:
Depending on whether or not you had the chance, you might be able to try give it a spin? Make a second Gamma DICOM script named "Gamma DICOM filter method"...